pkg/ctl/cmdutils/nodegroup_filter.go

```diff
@@ -72,70 +97,109 @@ func (f *NodeGroupFilter) ApplyExistingFilter(stackManager *manager.StackCollect
 		return err
 	}
 
-	f.existing.Insert(existing...)
+	f.ExcludeNames.Insert(existing...)
 
+	for _, name := range existing {
+		isAlsoIncluded := false
+		f.matchGlobs(name, f.includeGlobs, &isAlsoIncluded)
+		if isAlsoIncluded {
+			return fmt.Errorf("existing nodegroup %q should be excluded, but matches include fileter: %q", name, f.includeGlobsSpec)
+		}
+	}
 	return nil
 }
 
-// Match checks given nodegroup against the filter
-func (f *NodeGroupFilter) Match(ng *api.NodeGroup) bool {
-	if f.SkipAll {
-		return false
+func (*NodeGroupFilter) matchGlobs(name string, exprs []glob.Glob, result *bool) {
```

Comment:
Any particular reason to have the result as an argument instead of return value? I think it would be nicer if this method returned a bool result instead of having to pass an output argument. So it could be used like:


```if shouldInclude := f.matchGlobs(...); shouldInclude {...}```

Reply:
It just saves a few extra lines and a variable, e.g.:
```Go
mustInclude = f.includeNames.Has(name)
f.matchGlobs(name, f.includeGlobs, &mustInclude)
```

Reply:
Hm, I think this is a confusing API. Also is easier to read code when there's less state changes (so variables, when possible are assigned as fewer times as possible, typically just once).

Reply:
I agree, but it's just sooo many ifs... Will change.